src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -704,12 +704,12 @@ private void CheckStake(Block block, ChainedBlock chainTip)
             // Mark coinstake transaction.
             coinstakeContext.CoinstakeTx.Outputs.Add(new TxOut(Money.Zero, new Script()));
 
-            long balance = this.GetMatureBalance(utxoStakeDescriptions).Satoshi;
+            long balance = this.GetUtxoStakeDescriptionsSuitableForStaking(utxoStakeDescriptions, coinstakeContext.CoinstakeTx.Time, long.MaxValue).Sum(b => b.TxOut.Value);
```

Comment:
Before this change we used to get mature balance (mature balance is the balance we can spend, but some of utxos that are being used in it's calculation may be not suitable for staking because they dont meet the coinstake age maturity). 
This balance was needed in order to calculate target value for the `GetUtxoStakeDescriptionsSuitableForStaking`. 

With the new implementation you are likely to reserve more than is required. 
For example: we have 10 coins and 3 of them are confirmed just in a few blocks. In the current code those 3 coins are being cut off by the first `GetUtxoStakeDescriptionsSuitableForStaking` and after that you cut off some more coins for the reserve balance. But what we wanted to do is to just make sure that we have enough coins suitable for spending with the sum of reserve balance. 

So the logic effectively became incorrect after that optimization.